pkg/machinery/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -2605,6 +2605,30 @@ type FeaturesConfig struct {
 	//     Enable XFS project quota support for EPHEMERAL partition and user disks.
 	//     Also enables kubelet tracking of ephemeral disk usage in the kubelet via quota.
 	DiskQuotaSupport *bool `yaml:"diskQuotaSupport,omitempty"`
+	//   description: |
+	//     API server load balancer support - local proxy on defined port that will distribute
+	//     requests to all API servers in the cluster.
+	APIServerBalancerSupport *APIServerBalancer `yaml:"apiServerBalancerSupport,omitempty"`
+}
+
+// APIServerBalancer describes the configuration for the API server load balancer.
+type APIServerBalancer struct {
+	//   description: |
+	//     Enable API server load balancer support - will start local proxy.
+	ServerEnabled *bool `yaml:"enabled,omitempty"`
+	//   description: |
+	//     API server load balancer port.
+	ServerPort int `yaml:"port,omitempty"`
+}
+
+// Enabled implements config.APIServerBalancer.
+func (a *APIServerBalancer) Enabled() bool {
+	return pointer.SafeDeref(a.ServerEnabled)
+}
+
+// Port implements config.APIServerBalancer.
+func (a *APIServerBalancer) Port() int {
+	return a.ServerPort
```

Comment:
we should have a sane default here, otherwise setting to enabled makes it pick up port 0??

Reply:
Done - now 7445 port will be used as the default if `ServerPort` is zero.